src/datta_lab_to_nwb/markowitz_gillis_nature_2023/markowitz_gillis_nature_2023behaviorinterface.py

```diff
@@ -1,22 +1,108 @@
+                                     filters=[('uuid', '==', session_uuid)])
+        for col in self.source_data['metadata_columns']:
+            first_notnull = session_df.loc[session_df[col].notnull(), col].iloc[0]
+            metadata[col] = first_notnull
+        session_name = (
+                set(session_df.session_name[session_df.session_name.notnull()]) |
+                set(session_df.SessionName[session_df.SessionName.notnull()])
+                        )
+        assert len(session_name) == 1, "Multiple session names found"
+        metadata['session_name'] = session_name.pop()
         
         return metadata
 
-    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
-        # All the custom code to write to PyNWB
-        
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, session_uuid: str) -> NWBFile:
+        """Run conversion of data from the source file into the nwbfile."""
+        session_df = pd.read_parquet(self.source_data['filename'],
+                                     columns=self.source_data['columns'],
+                                     filters=[('uuid', '==', session_uuid)])
+
+        # Add Position Data
+        position_data = np.vstack((session_df.centroid_x_mm, session_df.centroid_y_mm, session_df.height_ave_mm)).T
+        position_spatial_series = SpatialSeries(
+            name="SpatialSeries",
+            description="Position (x, y, height) in an open field.",
+            data=position_data,
+            timestamps=session_df.timestamp.to_numpy(),
+            reference_frame=metadata['reference_frame'],
+            unit="mm",
+        )
+        position = Position(spatial_series=position_spatial_series)
+
+        # Add Compass Direction Data
+        direction_spatial_series = SpatialSeries(
+            name="SpatialSeries",
+            description="View angle of the subject measured in radians.",
+            data=session_df.angle_unwrapped.to_numpy(),
+            timestamps=session_df.timestamp.to_numpy(),
+            reference_frame=metadata['reference_frame'],
+            unit="radians",
+        )
+        direction = CompassDirection(
+            spatial_series=direction_spatial_series, name="CompassDirection"
+        )
+
+        # Add Syllable Data
+        syllable_time_series = TimeSeries(
+            name="syllable",
+            data=session_df["predicted_syllable (offline)"].to_numpy(),
+            timestamps=session_df.timestamp.to_numpy(),
+            description="Behavioral Syllable identified by Motion Sequencing (MoSeq).",
+            unit="None",
+        )
+        behavioral_time_series = BehavioralTimeSeries(
+            time_series=syllable_time_series,
+            name="SyllableTimeSeries",
+        )
+
+        # Combine all data into a behavioral processing module
+        behavior_module = nwbfile.create_processing_module(
+            name="behavior", description="Processed behavioral data"
+        )
```

Comment:
The `get_module` function from `neuroconv.tools.nwb_helpers` is recommended in case another interface has already added this module; otherwise an error would occur here